app/src/main/java/org/akvo/flow/presentation/settings/PreferenceActivity.java

```diff
@@ -207,44 +222,67 @@ void onDownloadFormOptionTap() {
             R.id.preference_reload_forms_subtitle
     })
     void onReloadAllSurveysOptionTap() {
+        if (trackingHelper != null) {
+            trackingHelper.logDownloadFormsPressed();
+        }
         DialogFragment newFragment = PassCodeReloadFormsDialog.newInstance();
         newFragment.show(getSupportFragmentManager(), PassCodeReloadFormsDialog.TAG);
     }
 
     @OnClick(R.id.preference_gps_fixes)
     void onGpsFixesTap() {
+        if (trackingHelper != null) {
+            trackingHelper.logGpsFixesEvent();
+        }
         navigator.navigateToGpsFixes(this);
     }
 
     @OnClick(R.id.preference_storage)
     void onCheckSdCardStateOptionTap() {
+        if (trackingHelper != null) {
+            trackingHelper.logStorageEvent();
+        }
         navigator.navigateToStorageSettings(this);
     }
 
     @OnCheckedChanged(R.id.switch_enable_data)
     void onDataCheckChanged(boolean checked) {
         if (trackChanges) {
+            if (trackingHelper != null) {
+                trackingHelper.logMobileDataChanged(checked);
+            }
             presenter.saveEnableMobileData(checked);
         }
     }
 
     @OnCheckedChanged(R.id.switch_screen_on)
     void onScreenOnCheckChanged(boolean checked) {
         if (trackChanges) {
+            if (trackingHelper != null) {
+                trackingHelper.logScreenOnChanged(checked);
+            }
             presenter.saveKeepScreenOn(checked);
         }
     }
 
     @OnItemSelected(R.id.preference_language)
     void onLanguageSelected(int position) {
         if (trackChanges) {
+            if (trackingHelper != null) {
+                trackingHelper.logLanguageChanged(languages.get(position));
+            }
             presenter.saveAppLanguage(position, languages);
         }
     }
 
     @OnItemSelected(R.id.preference_image_size)
     void onImageSizeSelected(int position) {
-        presenter.saveImageSize(position);
+        if (trackChanges) {
```

Comment:
Only here I see that the checking for whether tracking is enabled before logging an event. It isn't clear to me how that checking is handled in other places.

Reply:
Oh this had nothing to do with tracking, I have renamed it